gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -187,34 +235,97 @@ def handle_event(self, ev):
+            traceback.print_exc()
+            LOG.error("Failed to process ev.id=%s, ev=%s reason=%s" % (ev.id,
+                                                                       ev.data,
+                                                                       err))
+            result = FAILED
+
+        if ev.id == const.EVENT_CONFIGURE_HEALTHMONITOR:
+            if (kwargs.get('periodicity') == INITIAL and
+                    result == SUCCESS):
+                notification_data = self._prepare_notification_data(ev, result)
+                self._sc.poll_event_done(ev)
+                self._notification(notification_data)
+            elif kwargs.get('periodicity') == FOREVER:
+                if result == FAILED:
+                    """If health monitoring fails continuously for 5 times
+                       send fail notification to orchestrator
+                    """
+                    kwargs['fail_count'] = kwargs.get('fail_count') + 1
+                    if kwargs.get('fail_count') >= MAX_FAIL_COUNT:
+                        notification_data = self._prepare_notification_data(
+                                                                        ev,
+                                                                        result)
+                        self._sc.poll_event_done(ev)
+                        self._notification(notification_data)
+                elif result == SUCCESS:
+                    """set fail_count to 0 if it had failed earlier even once
+                    """
+                    kwargs['fail_count'] = 0
+        elif ev.id == const.EVENT_CLEAR_HEALTHMONITOR:
+            """Stop current poll event. event.key is vmid which will stop
+               that particular service vm's health monitor
+            """
+            notification_data = self._prepare_notification_data(ev, result)
+            self._sc.poll_event_done(ev)
+            self._notification(notification_data)
+        else:
+            """For other events, irrespective of result send notification"""
+            notification_data = self._prepare_notification_data(ev, result)
             self._notification(notification_data)
 
+    def _prepare_notification_data(self, ev, result):
+        kwargs = ev.data.get('kwargs')
+        request_info = kwargs['request_info']
+        del kwargs['request_info']
+        context = ev.data.get('context')
+
+        # Retrieve notification and remove it from context. Context is used
+        # as transport from batch processing function to this last event
+        # processing function. To keep the context unchanged, delete the
+        # notification_data before invoking driver API.
+        notification_data = context.get('notification_data')
+        del context['notification_data']
+        resource = context.get('resource')
+        del context['resource']
+
+        msg = {'receiver': const.ORCHESTRATOR,
+               'resource': resource,
+               'method': ev.id,
```

Comment:
When crossed checked this with the demo_26thfeb2016 branch. method name is same as ev.id. What should we do here ?